### PR TITLE
Update sections redux when the teacher adds/removes students

### PR DESCRIPTION
**Pull request by jmkulwik**
Previously, when a teacher would add or remove a student to a section in the "Manage Students" section of the teacher dashboard, it would cause a console error to be thrown when the teacher switched to the "Progress" section. This happened because the sectionData redux, where the class list exists, would only be set on page load and never later. 

Now, we add a call to setSection to set the sectionData with the updated class list any time a student is added, removed, or transferred via the teacher dashboard.

Potential concerns: When we call setSection, we set it using updated database information from a call to the server (rather than just setting it in the redux store from the client). However, since the data model of this data changes when saved, stored in redux, and retrieved from the server, this was the most fail-safe method without redesigning our teacher_dashboard redux store.

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by cforkish — CHANGES_REQUESTED** [review]
looking at this code always gives me so uncomfortable (not yours, just our redux stuff in general). that said, this looks like a good solution.

one minor change request: i don't think you want to merge your schema_cache.dump, do you?

**Review by clareconstantine — APPROVED** [review]
This seems like a good way to keep things up to date given the current messy state of this redux system.

**Comment by jmkulwik** [comment]
Force-push was a rebase onto staging

File: apps/test/unit/templates/manageStudents/ManageStudentsActionsCellTest.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import React from 'react';
+import $ from 'jquery';
+import sinon from 'sinon';
 import {shallow} from 'enzyme';
 import {expect} from '../../../util/deprecatedChai';
+import * as client from '@cdo/apps/util/userSectionClient';
+// import * as sectionData from '@cdo/apps/redux/sectionDataRedux';
```
Comment (molly-moen) [inline apps/test/unit/templates/manageStudents/ManageStudentsActionsCellTest.js]: remove this comment?

**Review by molly-moen — APPROVED** [review]

**Review by cforkish — APPROVED** [review]

File: apps/test/unit/templates/manageStudents/ManageStudentsActionsCellTest.js (hunk 1 above)
Comment (jmkulwik) [reply apps/test/unit/templates/manageStudents/ManageStudentsActionsCellTest.js]: Oh oops. Thank you!

**Pull request merged**